pkg/system/reconciler.go

```diff
@@ -286,6 +291,45 @@ func (r *Reconciler) Reconcile() (reconcile.Result, error) {
 	return res, nil
 }
 
+// VerifyObjectBucketCleanup checks if the uninstallation is in mode graceful and
+// if OBs still exist in the system the operator will wait
+// and the finalizer on noobaa CR won't be reomved
+func (r *Reconciler) VerifyObjectBucketCleanup() error {
+	log := r.Logger
+
+	if r.NooBaa.DeletionTimestamp != nil {
+		finalizersArray := r.NooBaa.GetFinalizers()
+
+		if util.Contains("noobaa.io/graceful_finalizer", finalizersArray) {
+
+			obcSelector, _ := labels.Parse("noobaa-domain=" + options.SubDomainNS())
+			objectBuckets := &nbv1.ObjectBucketList{}
+			util.KubeList(objectBuckets, &client.ListOptions{LabelSelector: obcSelector})
+
+			if len(objectBuckets.Items) == 0 {
+				log.Infof("All object buckets deleted in namespace %q", r.NooBaa.Namespace)
+				util.RemoveFinalizer(r.NooBaa, nbv1.GracefulFinalizer)
+				if !util.KubeUpdate(r.NooBaa) {
+					log.Errorf("NooBaa %q failed to remove finalizer %q", r.NooBaa.Name, nbv1.GracefulFinalizer)
+				}
+
+			} else {
+				var bucketNames []string
+				for i := range objectBuckets.Items {
+					ob := &objectBuckets.Items[i]
+					bucketName := ob.Spec.Endpoint.BucketName
```

Comment:
Why can't we print the ob.Name? it's not clearer for the user what's not deleted yet?

Reply:
it is printed in line 323 in an array of all the object bucket that were not deleted yet

Reply:
but why the name of the acutal buckets and not just ob.Name? that's what I'm missing...

Reply:
I checked and it's not the same name. After a discussion with Jacky, we agreed to change it to ob.Name.